samples/bluetooth/unicast_audio_client/src/main.c

```diff
@@ -277,7 +277,9 @@ static int init_lc3(void)
 
 	if (lc3_encoder == NULL) {
 		printk("ERROR: Failed to setup LC3 encoder - wrong parameters?\n");
+		return -1;
```

Comment:
I don't think it's wise to mix `-1` with POSIX error numbers, unless your intention is to `return -EPERM` here (EPERM is defined as `1` in our libc). Earlier in this function you're e.g. assigning the return value from `bt_audio_codec_cfg_freq_to_freq_hz()` to `ret`, and that public API can return a negative posix error.

Reply:
Hmm, I can see these audio samples actually do this kind of `-1` mixing with POSIX error codes in quite many places, so that deserves its own PR.

Reply:
This isn't an API function though, and just a local function where the error code is effectively just thrown away in the end. 

But otherwise I agree with you that we probably should try to stick to using errno values, even in cases like this

Reply:
Sure, I realize there's nothing critical being done with `ret`. But even then people look at sample code for reference, and this kind of bad design then propagates itself further. Anyway, I'll take back my "request for changes" since this needs to be fixed separately in a wider scope in a separate PR.